src/content/en/fundamentals/security/credential-management/retrieve-credentials.md

```diff
@@ -4,369 +4,396 @@ book_path: /web/fundamentals/_book.yaml
     });
 
+Google Sign-In results in an ID token as a proof of authentication,
+which gets sent to the server to create a session.
 
-Google Sign-In results in an id token as a proof of authentication which
-you send to the server to create a session.
+### Store identity information
 
-For additional identity providers, please refer to respective documentation:
+Once authentication is done, you can store the identity information. The information you’ll store here is the id from the identity provider and a provider string that represents the identity provider (`name` and `iconURL` are optional).
 
-* [Facebook](https://developers.facebook.com/docs/facebook-login)
-* [Twitter](https://dev.twitter.com/web/sign-in/implementing)
-* [GitHub](https://developer.github.com/v3/oauth/)
+To store federated account details, instantiate a new 
+[`FederatedCredential`](/web/fundamentals/security/credential-management/reference#federatedcredential), object with the user's identifier and the provider's identifier.
+Then invoke
+[`navigator.credentials.store()`](/web/fundamentals/security/credential-management/reference#navigatorcredentialstorecred) to store the identity information.
 
+For example:
 
+    // After a successful federation, instantiate a FederatedCredential
+    var cred = new FederatedCredential({
+      id:       id,                           // id in IdP
+      provider: 'https://account.google.com', // A string representing IdP
+      name:     name,                         // name in IdP
+      iconURL:  iconUrl                       // Profile image url
+    });
+
+    // Store it
+    navigator.credentials.store(cred)
+    .then(function() {
+      // continuation
+    });
 
 ## Sign out {: #sign-out }
 
-When a user signs out of your website, it's your responsibility to ensure
-that the user is not automatically signed in on their next visit. To turn off
-auto sign-in, call
+Sign out your users when the sign-out button is tapped.
+First terminate the session,
+then turn off auto sign-in for future visits.
+(How you terminate your sessions is totally up to you.)
+
+<!--Todo: replace this with new animation.
+<div class="attempt-right">
+  <figure>
+    <video src="animations/credential-management-smaller.mov" style="max-height: 400px;" autoplay muted loop controls></video>
+    <figcaption>Federated Login Flow</figcaption>
+  </figure>
+</div>-->
+
+### Turn off auto sign-in for future visits
+
+Call
 [`navigator.credentials.requireUserMediation()`](https://developer.mozilla.org/en-US/docs/Web/API/CredentialsContainer/requireUserMediation):
 
-    // After a user signing out...
-    navigator.credentials.requireUserMediation();
+    signoutUser();
+    if (navigator.credentials) {
+     navigator.credentials.requireUserMediation();
+    }
+
+This will ensure the auto sign-in won’t happen until next time the user enables auto sign-in. Users can do so by tapping on an account in an account chooser which is described in next section.
```

Comment:
I would remove the sentence "Users can ...." and "Then" in the next paragraph and concatenate them together.